test/mpi/coll/customreduce.c

```diff
@@ -0,0 +1,102 @@
+#include "mpi.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include "mpitest.h"
+
+int add_op(const void *invec, void *inoutvec, int *count, MPI_Datatype datatype)
+{
+    int i;
+    int *cin = (int *) invec, *cout = (int *) inoutvec;
+    for (i = 0; i < *count; i++) {
+        cout[i] = cin[i] + cout[i];
+    }
+    return 1;
+}
+
+/*
+ * Test case for custom reduce operator to be used with custom datatype
+ *
+ */
+int main(int argc, char *argv[])
+{
+    int rank, count, size, minsize = 1;
+    int isLeftGroup;
+    MPI_Status status;
+    unsigned int errs = 0;
+    MPI_Comm comm;
+    MTestDatatype sendtype, recvtype;
+    MTest_Init(&argc, &argv);
+
+    //Reduce with custom operator
```

Comment:
We try to avoid C++ style comments if we can. Please use `/* */ for consistency.